Sources/ResourceProvider/Cache/AnyAsyncCache.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  AnyAsyncCache.swift
+//  swift-resource-provider
+//
+//  Created by Óscar Morales Vivó on 4/11/23.
+//
+
+/**
+ Type-erased async cache.
+
+ This wrapper value type can be (and is) used to build up adapters for actual cache types, and can also be used to
+ build mocks for testing.
+ */
+public struct AnyAsyncCache<ID: Hashable, Value> {
+    /**
+     A type erased cache has its functionality injected as blocks.
+     - Parameters:
+       - valueForID: Block that implements `AsyncCache.valueFor(id:)`
+       - storeValueForID: Block that implements `AsyncCache.store(value:id:)`
+     */
+    public init(
+        valueForID: @escaping (ID) async -> Value? = { _ in nil },
+        storeValueForID: @escaping (Value, ID) async -> Void = { _, _ in }
+    ) {
+        self.valueForID = valueForID
+        self.storeValueForID = storeValueForID
+    }
+
+    /// Implements `AsyncCache.valueFor(id:)`
+    public let valueForID: (ID) async -> Value?
+
+    /// Implements `AsyncCache.store(value:id:)`
+    public let storeValueForID: (Value, ID) async -> Void
+}
+
+extension AnyAsyncCache: AsyncCache {
+    public func valueFor(id: ID) async -> Value? {
```

Comment:
The `For` reads weird to me, I would have spelled it as `value(for id: ID)`

Reply:
I'm of two minds about that. I know it's what the style recommends but while it reads better on declaration the direct parameter labeling reads clearer on use IMHO.

Reply:
i agree with dimitry valueFor reads more objc than swift

Reply:
Ok ok two against one ain't fair. Here's my lunch money…